music_assistant/providers/podcast-index/provider.py

```
@@ -0,0 +1,636 @@
+
+        if not self.api_key or not self.api_secret:
+            raise LoginFailed("API key and secret are required")
+
+        # Test API connection
+        try:
+            await self._api_request("stats/current")
+        except (LoginFailed, ProviderUnavailableError):
+            # Re-raise these specific errors as they have proper context
+            raise
+        except aiohttp.ClientConnectorError as err:
+            raise ProviderUnavailableError(
+                f"Failed to connect to Podcast Index API: {err}"
+            ) from err
+        except aiohttp.ServerTimeoutError as err:
+            raise ProviderUnavailableError(f"Podcast Index API timeout: {err}") from err
+        except Exception as err:
+            raise LoginFailed(f"Failed to connect to API: {err}") from err
+
+    async def search(
+        self, search_query: str, media_types: list[MediaType], limit: int = 10
+    ) -> SearchResults:
+        """
+        Perform search on Podcast Index.
+
+        Searches for podcasts by term. Future enhancement could include
+        category search if needed.
+        """
+        result = SearchResults()
+        if MediaType.PODCAST not in media_types:
+            return result
+
+        try:
+            response = await self._api_request(
+                "search/byterm", params={"q": search_query, "max": limit}
+            )
+
+            podcasts = []
+            for feed_data in response.get("feeds", []):
+                podcast = parse_podcast_from_feed(
+                    feed_data, self.lookup_key, self.domain, self.instance_id
+                )
+                if podcast:
+                    podcasts.append(podcast)
+
+            result.podcasts = podcasts
+
+        except (MediaNotFoundError, ProviderUnavailableError):
+            # Re-raise these specific errors
+            raise
+        except Exception as err:
+            self.logger.warning("Search failed: %s", err)
```

no need for this try...except block - just let the exception bubble up from the calling method

---

Fixed